src/components/tl/sharp/tl_sharp_team.c

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_sharp.h"
+#include "tl_sharp_coll.h"
+#include "core/ucc_mc.h"
+#include "core/ucc_ee.h"
+#include "coll_score/ucc_coll_score.h"
+
+UCC_CLASS_INIT_FUNC(ucc_tl_sharp_team_t, ucc_base_context_t *tl_context,
+                    const ucc_base_team_params_t *params)
+{
+    ucc_tl_sharp_context_t *ctx    =
+        ucc_derived_of(tl_context, ucc_tl_sharp_context_t);
+    struct sharp_coll_comm_init_spec comm_spec;
+    int ret;
+
+    UCC_CLASS_CALL_SUPER_INIT(ucc_tl_team_t, &ctx->super, params->team);
+
+    self->oob              = params->params.oob;
+    self->size             = self->oob.n_oob_eps;
+    self->rank             = params->rank;
+    self->oob_ctx.ctx      = self;
+    self->oob_ctx.ctx_oob  = NULL;
+    self->oob_ctx.team_oob = &self->oob;
+
+    comm_spec.rank              = self->rank;;
+    comm_spec.size              = self->size;
+    comm_spec.group_world_ranks = NULL;
```

Comment:
is there any advantage if we provide group_world_ranks? saves some allgather inside?